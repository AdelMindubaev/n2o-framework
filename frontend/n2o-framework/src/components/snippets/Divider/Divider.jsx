import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const dividerClass = props => {
  const { className, dashed, position, type, children } = props;
  return cx(className, {
    [`divider-h divider-h_with-title divider-h_${position} divider-h_dashed`]:
      type === 'horizontal' && children && dashed,
    [`divider-h divider-h_with-title divider-h_${position}`]:
      type === 'horizontal' && children && !dashed,
    'divider-h divider_horizontal divider-h_dashed':
      type === 'horizontal' && dashed && !children,
    'divider-h divider_horizontal':
      type === 'horizontal' && !dashed && !children,
    'divider-v divider_vertical divider-v_dashed':
      type === 'vertical' && dashed,
    'divider-v divider_vertical': type === 'vertical' && !dashed,
  });
};

/**
 * Компонент Divider
 * @reactProps {string} className - Пользовательский className css (default : '');
 * @reactProps {object} style - css стили Divider. Необходимые параметры width-для горизонтального, height-для вертикального (default : none);
 * @reactProps {boolean} dashed - вкл/откл путктирной линии на Divider (default : false);
 * @reactProps{string} position - Позиционирование заголовка. Работает только на горизонтальном Divider и при наличии заголовка (default : 'left');
 * @reactProps {string} type - Тип Divider horizontal или vertical (default : 'horizontal');
 * @reactProps {string} children - Заголовок для горизонтального типа. Указывается между тегами компонента: <Divider>Заголовок</Divider> (default : none);
 * **/

export const Divider = props => {
  const { style, children } = props;
  return (
    <div className={dividerClass({ ...props })} style={style}>
      {children && <span className="divider-h__title">{children}</span>}
    </div>
  );
};

Divider.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
  dashed: PropTypes.bool,
  position: PropTypes.string,
  type: PropTypes.string,
  children: PropTypes.string,
};

Divider.defaultProps = {
  dashed: false,
  position: 'left',
  type: 'horizontal',
};
