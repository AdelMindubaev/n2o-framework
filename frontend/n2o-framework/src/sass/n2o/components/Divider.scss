.horizontal-before-after-params {
  display: block;
  width: 90%;
  content: '';
  border-bottom: $border-default;
  bottom: 43%;
}

.#{$horizontal} {
  display: flex;
  justify-content: center;
  width: 100%;
}

.#{$vertical} {
  display: flex;
  justify-content: center;
  height: 100%;
}

.#{$horizontal}__title {
  width: 100%;
  position: relative;
  display: flex;
  flex-flow: row nowrap;
  white-space: nowrap;
}

/*Horizontal-type*/

.divider_horizontal:before {
  @extend .horizontal-before-after-params;
  width: 100%;
}

.#{$horizontal}_with-title {
  .#{$horizontal}__title {
    &:before {
      @extend .horizontal-before-after-params;
      margin-right: 15px;
      position: relative;
    }

    &:after {
      @extend .horizontal-before-after-params;
      margin-left: 15px;
      position: relative;
    }
  }
}

/*Vertical-type*/

.divider_vertical:before {
  display: block;
  height: 100%;
  content: '';
  border-left: $border-default;
}

.#{$horizontal}_left .#{$horizontal}__title:before, .#{$horizontal}_right .#{$horizontal}__title:after {
  width: 10%;
}

.#{$horizontal}_dashed {
  .#{$horizontal}__title {
    &:before, &:after {
      border-bottom: $border-dashed;
    }
  }

  &:before {
    border-bottom: $border-dashed;
  }
}

.#{$vertical}_dashed {
  &:before {
    border-left: $border-dashed;
  }
}