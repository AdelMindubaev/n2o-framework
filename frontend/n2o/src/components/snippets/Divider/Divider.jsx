import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const dividerClass = props => {
  const { className, dashed, position, type, children } = props;
  return cx(className, {
    [`divider-h divider-h_with-title divider-h_${position} divider-h_dashed`]:
      type === 'horizontal' && children && dashed,
    [`divider-h divider-h_with-title divider-h_${position}`]:
      type === 'horizontal' && children && !dashed,
    'divider-h divider_horizontal divider-h_dashed':
      type === 'horizontal' && dashed && !children,
    'divider-h divider_horizontal':
      type === 'horizontal' && !dashed && !children,
    'divider-v divider_vertical divider-v_dashed':
      type === 'vertical' && dashed,
    'divider-v divider_vertical': type === 'vertical' && !dashed,
  });
};

/**
 * Компонент Divider
 * @param {string} className - Пользовательский className css (default : '');
 * @param {object} style - css стили Divider. Необходимые параметры width-для горизонтального, height-для вертикального (default : none);
 * @param {boolean} dashed - вкл/откл путктирной линии на Divider (default : false);
 * @param {string} position - Позиционирование заголовка. Работает только на горизонтальном Divider и при наличии заголовка (default : 'left');
 * @param {string} type - Тип Divider horizontal или vertical (default : 'horizontal');
 * @param {string} children - Заголовок для горизонтального типа. Указывается между тегами компонента: <Divider>Заголовок</Divider> (default : none);
 * **/

export const Divider = props => {
  const { style, children } = props;
  return (
    <div className={dividerClass({ ...props })} style={style}>
      {children && <span>{children}</span>}
    </div>
  );
};

Divider.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
  dashed: PropTypes.bool,
  position: PropTypes.string,
  type: PropTypes.string,
  children: PropTypes.string,
};

Divider.defaultProps = {
  dashed: false,
  position: 'left',
  type: 'horizontal',
};
