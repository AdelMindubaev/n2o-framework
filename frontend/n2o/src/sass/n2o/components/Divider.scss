.divider-h {
  display: flex;
  justify-content: center;
  width: 100%;
}

.divider-v {
  display: flex;
  justify-content: center;
  height: 100%;
}

span {
  width: 100%;
  position: relative;
  display: flex;
  flex-flow: row nowrap;
  white-space: nowrap;
}

/*Horizontal-type*/

.divider_horizontal:before {
  display: block;
  width: 100%;
  content: '';
  border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  bottom: 43%;
}

.divider-h_with-title span {
  &:before {
    display: block;
    margin-right: 15px;
    width: 90%;
    content: '';
    position: relative;
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    bottom: 43%;
  }

  &:after {
    display: block;
    margin-left: 15px;
    width: 90%;
    content: '';
    position: relative;
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    bottom: 43%;
  }
}

/*Vertical-type*/

.divider_vertical:before {
  display: block;
  height: 100%;
  content: '';
  border-left: 1px solid rgba(0, 0, 0, 0.4);
}

.divider-h_left span:before, .divider-h_right span:after {
  width: 10%;
}

.divider-h_dashed {
  span {
    &:before, &:after {
      border-bottom: 1px dashed rgba(0, 0, 0, 0.4);
    }
  }

  &:before {
    border-bottom: 1px dashed rgba(0, 0, 0, 0.4);
  }
}

.divider-v_dashed:before {
  border-left: 1px dashed rgba(0, 0, 0, 0.4);
}