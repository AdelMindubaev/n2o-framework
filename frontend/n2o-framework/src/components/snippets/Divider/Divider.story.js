import React from 'react';
import { storiesOf } from '@storybook/react';
import { Divider } from './Divider';

const stories = storiesOf('UI Компоненты/Divider', module);

const DividerToStory = props => {
  return <Divider {...props} />;
};

stories.add('Компонент', () => {
  return <DividerToStory />;
});

stories.add('Горизонтальный', () => {
  return <DividerToStory type={'horizontal'} />;
});

stories.add('Пунктирная линия', () => {
  return <DividerToStory dashed={true} />;
});

stories.add('С заголовком (по умолчанию лево)', () => {
  return <DividerToStory children={'Divider title'} />;
});

stories.add('Пунктирна линия, с заголовком (по умолчанию лево)', () => {
  return <DividerToStory dashed={true} children={'Divider title'} />;
});

stories.add('С заголовком (центр)', () => {
  return <DividerToStory position={'center'} children={'Divider title'} />;
});

stories.add('С заголовком (право)', () => {
  return <DividerToStory position={'right'} children={'Divider title'} />;
});

stories.add('Вертикальный, высота 250px', () => {
  return <DividerToStory type={'vertical'} style={{ height: '250px' }} />;
});

stories.add('Вертикальный, пунктирная линия, высота 250px', () => {
  return (
    <DividerToStory
      dashed={true}
      type={'vertical'}
      style={{ height: '250px' }}
    />
  );
});

stories.add('Стилизация цвета заголовка', () => {
  return (
    <section>
      <div style={{ margin: '30px' }}>
        <DividerToStory
          style={{ width: '100%', color: '#b4004e' }}
          children={'Divider title'}
        />
      </div>
      <div style={{ margin: '30px' }}>
        <DividerToStory
          style={{ width: '100%', color: '#00766c' }}
          children={'Divider title'}
          position={'center'}
        />
      </div>
      <div style={{ margin: '30px' }}>
        <DividerToStory
          style={{ width: '100%', color: '#8eacbb' }}
          children={'Divider title'}
          position={'right'}
        />
      </div>
    </section>
  );
});

stories.add('Различная ширина (style)', () => {
  return (
    <section>
      <div style={{ margin: '30px' }}>
        <DividerToStory style={{ width: '25%' }} children={'25%'} />
      </div>
      <div style={{ margin: '30px' }}>
        <DividerToStory style={{ width: '50%' }} children={'50%'} />
      </div>
      <div style={{ margin: '30px' }}>
        <DividerToStory style={{ width: '75%' }} children={'75%'} />
      </div>
      <div style={{ margin: '30px' }}>
        <DividerToStory style={{ width: '100%' }} children={'100%'} />
      </div>
    </section>
  );
});
