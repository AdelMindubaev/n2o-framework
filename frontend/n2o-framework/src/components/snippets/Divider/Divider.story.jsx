import React from 'react'
import { storiesOf } from '@storybook/react'

import { Divider } from './Divider'

const stories = storiesOf('UI Компоненты/Divider', module)

const DividerToStory = props => <Divider {...props} />

stories.add('Компонент', () => <DividerToStory />)

stories.add('Горизонтальный', () => <DividerToStory type="horizontal" />)

stories.add('Пунктирная линия', () => <DividerToStory dashed />)

stories.add('С заголовком (по умолчанию лево)', () => <DividerToStory children="Divider title" />)

stories.add('Пунктирна линия, с заголовком (по умолчанию лево)', () => <DividerToStory dashed children="Divider title" />)

stories.add('С заголовком (центр)', () => <DividerToStory position="center" children="Divider title" />)

stories.add('С заголовком (право)', () => <DividerToStory position="right" children="Divider title" />)

stories.add('Вертикальный, высота 250px', () => <DividerToStory type="vertical" style={{ height: '250px' }} />)

stories.add('Вертикальный, пунктирная линия, высота 250px', () => (
    <DividerToStory
        dashed
        type="vertical"
        style={{ height: '250px' }}
    />
))

stories.add('Стилизация цвета заголовка', () => (
    <section>
        <div style={{ margin: '30px' }}>
            <DividerToStory
                style={{ width: '100%', color: '#b4004e' }}
                children="Divider title"
            />
        </div>
        <div style={{ margin: '30px' }}>
            <DividerToStory
                style={{ width: '100%', color: '#00766c' }}
                children="Divider title"
                position="center"
            />
        </div>
        <div style={{ margin: '30px' }}>
            <DividerToStory
                style={{ width: '100%', color: '#8eacbb' }}
                children="Divider title"
                position="right"
            />
        </div>
    </section>
))

stories.add('Различная ширина (style)', () => (
    <section>
        <div style={{ margin: '30px' }}>
            <DividerToStory style={{ width: '25%' }} children="25%" />
        </div>
        <div style={{ margin: '30px' }}>
            <DividerToStory style={{ width: '50%' }} children="50%" />
        </div>
        <div style={{ margin: '30px' }}>
            <DividerToStory style={{ width: '75%' }} children="75%" />
        </div>
        <div style={{ margin: '30px' }}>
            <DividerToStory style={{ width: '100%' }} children="100%" />
        </div>
    </section>
))
