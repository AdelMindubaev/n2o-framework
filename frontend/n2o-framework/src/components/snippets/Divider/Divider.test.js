import React from 'react'
import { mount, shallow } from 'enzyme/build'
import renderer from 'react-test-renderer'

import { Divider } from './Divider'

const setup = (propOverrides) => {
    const props = {
        children: 'title',
        // use this to assign some default props

        ...propOverrides,
    }

    const wrapper = mount(<Divider {...props} />)

    return wrapper
}

describe('<Divider />', () => {
    it('проверяет создание элемента Divider', () => {
        const wrapper = shallow(<Divider />)
        expect(wrapper.find('.divider-h').exists()).toBeTruthy()
        expect(wrapper.find('.divider_horizontal').exists()).toBeTruthy()
    })
    it('проверяет наличие children', () => {
        const wrapper = setup()
        expect(wrapper.find('span').exists()).toBeTruthy()
    })
    it('проверка на соответствие заголовка = title', () => {
        const wrapper = setup()
        expect(wrapper.text()).toBe('title')
    })
    it('renders correctly', () => {
    // eslint-disable-next-line no-undef
        const tree = renderer
            .create(<Divider style={undefined}>title</Divider>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
