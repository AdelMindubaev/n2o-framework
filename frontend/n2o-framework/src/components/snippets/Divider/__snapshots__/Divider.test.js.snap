// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> renders correctly 1`] = `
<div
  className="divider-h divider-h_with-title divider-h_left"
>
  <span
    className="divider-h__title"
  >
    title
  </span>
</div>
`;
